dsa-23au/java-dsa/arrays-links/src/test/java/dev/codewithfriends/ArrayWrapperTest.java

```diff
@@ -9,15 +9,50 @@
 /**
  * Unit test for simple App.
  */
+
 public class ArrayWrapperTest
 {
+    String[] s1 = {"Bread","Eggs","Peanut Butter","Butter"};
+    @Test
+    public void testAdd(){
+        //Create list
+        List<String> l = new ArrayWrapper<String>(10);
+        for (int i = 0; i < s1.length; i++) {
+            l.add(s1[i]);
+            assertEquals(s1[i], l.get(i));
+        }
+        //Insert Broccoli at index 2 in l, k=2
+        l.add(2,"Broccoli");
+        int k = 2;
+        assertEquals("Bread", l.get(0));
+        assertEquals("Eggs", l.get(1));
+        assertEquals("Broccoli", l.get(k));
+        assertEquals("Peanut Butter", l.get(k+1));
+        assertEquals("Butter", l.get(k+2));
+        l.set(2, "Cauliflower");
+        assertEquals("Bread", l.get(0));
+        assertEquals("Eggs", l.get(1));
+        assertEquals("Cauliflower", l.get(k));
+        assertEquals("Peanut Butter", l.get(k+1));
+        assertEquals("Butter", l.get(k+2));
+    }
+
     Book b1 = new Book("Some title", "some author");
+
     @Test
-    public void insertShifts()
-    {
-        List l = new ArrayWrapper<Book>(10);
-        assertEquals(0, l.size());
-        l.add(b1);
-        assertEquals(1, l.size());
+    public void testSize(){
+        assertEquals(null, 0, 0, 0);
```

Comment:
For this method about testing the size, try to come up with an `assertEquals` statement that depends on `this.currentSize` and asserts that a newly created list has size 0, and after some inserts and deletes, has the appropriate new size.

This PR is already merged, so I'll open a new issue to describe this work.